scripts/classificationloss.py

```diff
@@ -0,0 +1,220 @@
+import sys, os
+
+from PIL import Image
+from podm import metrics
+import numpy as np
+
+# Accroding to: https://pyimagesearch.com/2022/05/02/mean-average-precision-map-using-the-coco-evaluator/
+# This ia also helpful: https://github.com/yfpeng/object_detection_metrics/tree/master/src/podm
+
+
+def IOU(boxes1, boxes2):
+    x11, y11, x12, y12 = np.split(boxes1, 4, axis=1)
+    x21, y21, x22, y22 = np.split(boxes2, 4, axis=1)
+    xA = np.maximum(x11, np.transpose(x21))
+    yA = np.maximum(y11, np.transpose(y21))
+    xB = np.minimum(x12, np.transpose(x22))
+    yB = np.minimum(y12, np.transpose(y22))
+    interArea = np.maximum((xB - xA), 0) * np.maximum((yB - yA), 0)
+    boxAArea = (x12 - x11) * (y12 - y11)
+    boxBArea = (x22 - x21) * (y22 - y21)
+    iou = interArea / (boxAArea + np.transpose(boxBArea) - interArea)
+    return iou
+
+
+def boxes_xywh_to_xyxy(boxes):
+    res = np.copy(boxes[:, :])
+    res[:, 2] += res[:, 0]
+    res[:, 3] += res[:, 1]
+    return res
+
+
+def box_xywh_to_xyxy(boxes):
+    res = np.copy(boxes)
+    res[2] += res[0]
+    res[3] += res[1]
+    return res
+
+
+# Print iterations progress
+def printProgressBar(
+    iteration,
+    total,
+    prefix="",
+    suffix="",
+    decimals=1,
+    length=100,
+    fill="█",
+    printEnd="\r",
+):
+    """
+    Call in a loop to create terminal progress bar
+    @params:
+        iteration   - Required  : current iteration (Int)
+        total       - Required  : total iterations (Int)
+        prefix      - Optional  : prefix string (Str)
+        suffix      - Optional  : suffix string (Str)
+        decimals    - Optional  : positive number of decimals in percent complete (Int)
+        length      - Optional  : character length of bar (Int)
+        fill        - Optional  : bar fill character (Str)
+        printEnd    - Optional  : end character (e.g. "\r", "\r\n") (Str)
+    """
+    percent = ("{0:." + str(decimals) + "f}").format(100 * (iteration / float(total)))
+    filledLength = int(length * iteration // total)
+    bar = fill * filledLength + "-" * (length - filledLength)
+    print(f"\r{prefix} |{bar}| {percent}% {suffix}", end=printEnd)
+    # Print New Line on Complete
+    if iteration == total:
+        print()
```

Comment:
I'm sure there are several existing options for this so we don't have to maintain this. Maybe the podm lib already even does this?